Marlin/src/inc/SanityCheck.h

```diff
@@ -4507,6 +4507,10 @@ static_assert(WITHIN(MULTISTEPPING_LIMIT, 1, 128) && IS_POWER_OF_2(MULTISTEPPING
   #error "Only enable ULTIPANEL_FEEDMULTIPLY or ULTIPANEL_FLOWPERCENT, but not both."
 #endif
 
+#if ENABLED(CONFIGURABLE_MACHINE_NAME) && DISABLED(GCODE_QUOTED_STRINGS)
+  #error "CONFIGURABLE_MACHINE_NAME requires GCODE_QUOTED_STRINGS."
```

Comment:
May be nice to add this requirement to the base configuration.h

Reply:
I can add a comment to that effect. But since `GCODE_QUOTED_STRINGS` (`Configuration_adv.h`) follows `CONFIGURABLE_MACHINE_NAME` (`Configuration.h`) it won't be enforced with an `#if` in `Configuration.h`.